### PR TITLE
docs: update post-upgrading in upgrade page

### PR DESCRIPTION
**Pull request by dengmingtong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

update docs for post upgrading actions
Screenshot 2024-08-21 at 10 04 57

Screenshot 2024-08-21 at 10 04 14

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Comment by aws-solution-clickstream-analytics** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1568)
###### _updated: 8/17/2024, 04:05:28 (UTC+0)_

**Comment by aws-solution-clickstream-analytics** [comment]
# SonarQube Code Analytics
## Quality Gate passed

passed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-1568&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-1568&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-1568&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

File: docs/en/upgrade.md [hunk 1]
```diff
@@ -153,6 +153,15 @@ The solution automatically and asynchronously upgrades the views and materialize
     DROP PROCEDURE "<app-id>".sp_clear_item_and_user();
     ```
 
+## Post-Upgrade Actions II (only applicable for upgrading to V1.2.0 from earlier version)
```
Comment (zxkane) [inline docs/en/upgrade.md:156]: V1.2.0 -> v1.2..0

keep consistency in the doc

File: docs/en/upgrade.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ You can view the status of the stack in the AWS CloudFormation console in the **
 
 You can view the status of the pipeline in the solution console in the **Status** column. After a few minutes, you can receive an Active status.
 
-## Post-Upgrade Actions (only applicable to data modeling enabled)
+## Post-Upgrade Actions I (only applicable to data modeling enabled)
```
Comment (zxkane) [inline docs/en/upgrade.md:58]: It'd better for adding modules as sub-header, like,

```markdown
### Ingestion Module
...

### Data Modeling

### Upgrade the Data Schema and Out-of-the-box Dashboards

...
```

**Review by zxkane — CHANGES_REQUESTED** [review]

File: docs/en/upgrade.md (hunk 2 above)
Comment (dengmingtong) [reply docs/en/upgrade.md]: Refactored

File: docs/en/upgrade.md (hunk 1 above)
Comment (dengmingtong) [reply docs/en/upgrade.md]: Refactored

**Comment by aws-solution-clickstream-analytics** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1568)
###### _updated: 8/19/2024, 07:06:04 (UTC+0)_

File: docs/en/upgrade.md [hunk 3]
```diff
@@ -55,7 +55,15 @@ You can view the status of the stack in the AWS CloudFormation console in the **
...
+
+Due to AWS no longer supporting Launch Configurations in Auto Scaling Groups, we will use Launch Template instead. After upgrading the data ingestion module to v1.2.0, please follow below steps to replace the EC2 instances used by Amazon ECS with the new Launch Template configuration:
+
+1. Manually increase the desired number of tasks in Amazon ECS.
+2. Once the newly added Amazon ECS tasks have started successfully, manually stop the old tasks.
+3. Manually terminate the old EC2 instances.
+
+### Data Modeling
 
 ### Upgrade the Data Schema and Out-of-the-box Dashboards
```
Comment (zxkane) [inline docs/en/upgrade.md:68]: update to level 4 for Upgrade the Data Schema and Out-of-the-box Dashboards / Migrate the Existing Data (only applicable when upgrading from a version earlier than v1.1.6)

**Review by zxkane — CHANGES_REQUESTED** [review]

**Comment by aws-solution-clickstream-analytics** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1568)
###### _updated: 8/20/2024, 08:29:21 (UTC+0)_

File: docs/en/upgrade.md [hunk 4]
```diff
@@ -55,13 +55,21 @@ You can view the status of the stack in the AWS CloudFormation console in the **
 
 You can view the status of the pipeline in the solution console in the **Status** column. After a few minutes, you can receive an Active status.
 
-## Post-Upgrade Actions (only applicable to data modeling enabled)
```
Comment (zxkane) [inline docs/en/upgrade.md:58]: We still need to `## Post-Upgrade Actions`.

**Comment by aws-solution-clickstream-analytics** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1568)
###### _updated: 8/21/2024, 02:35:25 (UTC+0)_

**Review by zxkane — APPROVED** [review]

**Pull request merged**